annotation-utils/src/main/java/edu/isi/vista/annotationutils/PushAnnotations.kt

```diff
@@ -0,0 +1,177 @@
+     *          </li>
+     *     <li> `inceptionPassword` is the password for the user above. Note that parameter files can
+     *          interpolate from environmental variables </li>
+     *     <li> `exportedAnnotationRoot` is the directory to download annotation to.  The
+     *     hierarchy beneath it will look like "projectName/documentName/documentName-userName.json" </li>
+     *     <li> `restoreJson`: "yes" if you would like to restore the original document text to the annotation
+     *     files; if your repository is public, please put "no"; if "yes", you must also provide
+     *     the following four parameter values:
+     *       <ul>
+     *         <li> `indexDirectory` is the location of the files produced by `indexFlatGigaword.java`
+    (https://github.com/isi-vista/nlp-util/blob/master/nlp-core-open/src/main/java/edu/isi/nlp/corpora/gigaword/IndexFlatGigaword.java) </li>
+     *         <li> `gigawordDataDirectory` is the location of the gigaword text files </li>
+     *         <li> `inputJsonDirectory` is the location of the stripped json files produced by `ExportAnnotations.kt` </li>
+     *         <li> `restoredJsonDirectory` is where the new json files will go </li>
+     *       </ul>
+     *     </li>
+     *     <li> `repoToPushTo`: the ssh url of the repository to which the annotation data
+     *     will be pushed; for ISI's curated training work, this is
+     *     git@github.com:isi-vista/curated-training-annotation.git,
+     *     but this will differ for other projects </li>
+     *     <li> `localWorkingCopyDirectory`: the location of the local repository; the repository
+     *     specified by `repoToPushTo` will be cloned to this location if it does not exist here</li>
+     * </ul>
+     *
+     * Precondition: Your ssh private key is saved in ~/.ssh/id_rsa.
+     * Postcondition: The exported annotations will be saved within a folder named `data/` in the local
+     * working directory.
+     */
+    if (argv.size != 1) {
+        throw RuntimeException("Expected a single argument: a parameter file")
+    }
+    val paramsLoader = SerifStyleParameterFileLoader.Builder()
+            .interpolateEnvironmentalVariables(true).build()
+    val params = paramsLoader.load(File(argv[0]))
+    val exportAnnotationsParamsBuilder = Parameters.builder()
+    exportAnnotationsParamsBuilder.set("inceptionUrl", params.getString("inceptionUrl"))
+    exportAnnotationsParamsBuilder.set("inceptionUsername", params.getString("inceptionUsername"))
+    exportAnnotationsParamsBuilder.set("inceptionPassword", params.getString("inceptionPassword"))
+    exportAnnotationsParamsBuilder.set("exportedAnnotationRoot", params.getString("exportedAnnotationRoot"))
+    val exportAnnotationsParams = exportAnnotationsParamsBuilder.build()
+
+    val restoreJson = params.getString("restoreJson") == "yes"
+    val restoreJsonParamsBuilder = Parameters.builder()
+    if (restoreJson) {
```

Comment:
A cleaner way to do this:

```kotlin
val restoreJsonParameters = if (params.getOptionalBoolean("restoreOriginalText").orElse(false)) {
   Parameters.builder()
        set("indexDirectory", params.getString("indexDirectory"))
       .set("gigawordDataDirectory", params.getString("gigawordDataDirectory"))
       .set("inputJsonDirectory", params.getString("inputJsonDirectory"))
       .set("restoredJsonDirectory", params.getString("restoredJsonDirectory"))
       .build()
} else {
   null
}
```

Then below you can do `if (restoreJsonParams != null) ...`

Also note in the snippet above suggested changes to the parameter name and default value.  Since it is now a boolean/ it will have values `true`/`false` rather than `yes`/`no`.